keras_nlp/models/gpt2/gpt2_causal_lm.py

```diff
@@ -293,10 +294,24 @@ def generate(
                 "`preprocessor` is set before calling `generate`."
             )
         sampler = keras_nlp.samplers.get(sampler)
-        # `jit_compile` is a property of keras.Model after tf 2.12.
-        # Use `getattr()` for backwards compatibility.
-        sampler.jit_compile = getattr(self, "jit_compile", True)
+        if sampler.__class__.__name__ == "BeamSampler":
+            raise ValueError(
+                "`BeamSampler` is not supported right now, please choose "
+                "another sampler, e.g., `TopPSampler`."
+            )
+        if hasattr(self, "jit_compile"):
+            # `jit_compile` is a public property as of tf 2.12. hasattr is for
+            # backward compat.
+            sampler.jit_compile = self.jit_compile
         sampler.run_eagerly = self.run_eagerly
+        if hasattr(self, "sampler") and serialize(sampler) == serialize(
```

Comment:
just do `self.sampler = None` in init. we should avoid hasattr checking with objects under our control

Reply:
sg!